event-digger/src/App.js

```diff
@@ -55,10 +60,16 @@ class App extends Component {
 	  }, this.saveNotes);
 	};
   
-	editNote = (note) => {
+	editNote = (id) => {
+		console.log("hi")
 	  this.setState((state) => {
 		return {
-		  notes: state.notes.map((n) => (n.id === note.id ? note : n)),
+		  notes: state.notes.map((n) => {
+			if (n.id === id) {
+			  return {...n, count: n.count+1};
+			}
+			return n;
```

Comment:
Could you explain what this part of the code does?, why do return the count+1 if the `id`s are identical?